File: examples/example_maxwell_canon.ts
/** Smoke example: ideal regime + projector + Maxwell stack. */
const sst = require('../index.js');
if (!sst || !sst.available) {
  console.log('[SKIP] native addon unavailable');
  process.exit(0);
}
console.log('evaluateWorldsheetGuards', sst.evaluateWorldsheetGuards && sst.evaluateWorldsheetGuards(3, 1.0, 1e-8, false, false));
console.log('evaluateIdealKnotRegime', sst.evaluateIdealKnotRegime && sst.evaluateIdealKnotRegime(1, 1, 1, 1, 0, 0));
console.log('projectorSphereIntegral', sst.projectorSphereIntegral && sst.projectorSphereIntegral());
console.log('leadingResponseR0', sst.leadingResponseR0 && sst.leadingResponseR0(16.3714672385));
console.log('maxwellThreeGate', sst.maxwellThreeGate && sst.maxwellThreeGate(1, 2, 1, 0.1, 1));
console.log('evaluateMechanicalFalsifier', sst.evaluateMechanicalFalsifier && sst.evaluateMechanicalFalsifier(2, 1, 7e-7, 1));
console.log('evaluateSwirlTonic', sst.evaluateSwirlTonic && sst.evaluateSwirlTonic([1],[0],[0],[1],[0],[0], 1, false));

File: examples/example_maxwell_canon.py
"""Smoke example: ideal regime + transverse projector + Maxwell stack."""
from __future__ import annotations
import math, os, sys
sys.path.insert(0, os.path.abspath("."))
sys.path.insert(0, os.path.join(os.path.abspath("."), "src"))
try:
    from SSTcore import (
        IdealKnotRegimeAPI, TransverseProjectorAPI, SpectroResponseAPI,
        MaxwellKineticAPI, MechanicalFalsifierAPI, SwirlTonicAPI, WorldsheetGuardsAPI,
    )
except ImportError:
    from sstcore import (
        IdealKnotRegimeAPI, TransverseProjectorAPI, SpectroResponseAPI,
        MaxwellKineticAPI, MechanicalFalsifierAPI, SwirlTonicAPI, WorldsheetGuardsAPI,
    )

print("worldsheet", WorldsheetGuardsAPI.evaluate(3, 1.0, 1e-8, False, False).passed)
print("regime", IdealKnotRegimeAPI.evaluate(1.0, 1.0, 1.0, 1.0, 0.0, 0.0).regime)
print("8pi/3", TransverseProjectorAPI.projector_sphere_integral())
print("R0", TransverseProjectorAPI.leading_response_R0(TransverseProjectorAPI.high_res_ld()))
print("spectro", SpectroResponseAPI.evaluate(2.0, 1.0, 1.0, [1.0], [0.1], False).delta_nu)
print("three_gate", MaxwellKineticAPI.three_gate_condition(1.0, 2.0, 1.0, 0.1, 1.0))
print("falsifier", MechanicalFalsifierAPI.evaluate(2.0, 1.0, 7e-7, 1.0).C_blind)
print("tonic", SwirlTonicAPI.evaluate([1.0],[0.0],[0.0],[1.0],[0.0],[0.0], 1.0, False).holonomy)
